### PR TITLE
BSPIMX8M-3405 Wip/bhahn/rs485 update

### PR DESCRIPTION
**Pull request by BHahn42618**
Update rs485 chapter to have much more detailed documentation on how to test rs485.

### CODE REVIEW

File: source/bsp/peripherals/rs485-fullduplex.rsti [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+Also here you can do the echo test to see if sending and receiving works:
+
+.. code-block:: console
+   :substitutions:
+
+   target1:~$ cat /dev/ttymxc1
+   target2:~$ echo test > /dev/ttymxc1
+
+You should see "test" printed out on target1. You can also switch the roles and send on target2
```
Comment (jonas-rem) [inline source/bsp/peripherals/rs485-fullduplex.rsti:33]: Looks like a textwidth of more than 80 character. Can you re-format to max 80 characters wherever possible?

**Review by jonas-rem — CHANGES_REQUESTED** [review]

File: source/bsp/peripherals/rs485-fullduplex.rsti [hunk 2]
```diff
@@ -0,0 +1,55 @@
+RS485 full-duplex
+~~~~~~~~~~~~~~~~~
+
+For full-duplex operation, the wiring needs to be crossed not straight like in half-duplex mode.
+To see which pin has which function check the hardware manual.
```
Comment (ymoog) [inline source/bsp/peripherals/rs485-fullduplex.rsti:5]: This is not true. For RS232, the same principle applies.
Also, please choose more technical terms than "crossed" and "straight" (you don't have to replace the terms, you can rephrase which I think is more appropriate here).
Note that you need to differentiate between master and slave device. When you have multiple masters, then both their tx+ are connected to a slaves rx+ (For simple 2 device setups you always have master and slave).

File: source/bsp/peripherals/rs485-fullduplex.rsti [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+   ...
+   /dev/ttymxc1: count for this session: rx=114660, tx=118755, rx err=0
+   target2:~$ linux-serial-test -s -e -f -p /dev/ttymxc1 -b 115200 --rs485 0 -o 10 -i 15 -W 2
+   ...
+   /dev/ttymxc1: count for this session: rx=118755, tx=114660, rx err=0
+
+In this example both targets will send and receive simultaneously. They will receive for 15sec and
+send for 10sec. The receiver needs to receive a bit longer, so that all sent messages will get
+received. Remember to start both targets almost simultaneously. I did account for a small
+difference in time with the ``-W 2`` option. At the end the program will print the final
```
Comment (ymoog) [inline source/bsp/peripherals/rs485-fullduplex.rsti:51]:
```suggestion
received. Remember to start both targets almost simultaneously. A small
difference in start time is accounted for with the ``-W 2`` option. At the end the program will print the final
```

File: source/bsp/peripherals/rs485-fullduplex.rsti [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+"count for this session". There you can check, that all transmitted frames were received.
```
Comment (ymoog) [inline source/bsp/peripherals/rs485-fullduplex.rsti:52]: No comma here.

File: source/bsp/peripherals/rs485.rsti [hunk 5]
```diff
@@ -1,6 +1,9 @@
 RS485
 .....
 
+Remember to use bus termination resistors of 120 Ohm at each end of the bus,
+when using longer cables.
+
```
Comment (ymoog) [inline source/bsp/peripherals/rs485.rsti:6]: Put this as ``note`` or ``hint`` ?

**Review by ymoog — CHANGES_REQUESTED** [review]

File: source/bsp/peripherals/rs485-fullduplex.rsti (hunk 1 above)
Comment (BHahn42618) [reply source/bsp/peripherals/rs485-fullduplex.rsti:33]: Ah ok, I formatted to 100 characters per line (like in ptf). Will do

File: source/bsp/peripherals/rs485-fullduplex.rsti (hunk 2 above)
Comment (BHahn42618) [reply source/bsp/peripherals/rs485-fullduplex.rsti]: I have removed this passage and put another picture on how to do the wiring here instead.

**Review by ymoog — APPROVED** [review]

**Pull request merged**